src/states/proving_node.rs

```diff
@@ -293,6 +377,9 @@ impl Base for ProvingNode {
                     difficulty,
                     log_ident,
                 );
+                if let Some(status) = self.resource_proofing_status.get_mut(&pub_id) {
```

Comment:
Would we be better off using a BTreeSet?

Reply:
we need the status value to decide whether to send out CandidateInfo or not, so I think a map is required?

Reply:
Yes, we are populating the map initially and iterating over it. make sense,=.